### PR TITLE
Overlay on disabled elements, to catch events and show tooltips

### LINKED ISSUE #27451: Tooltip component does not work with disabled Button component
**Describe the bug**

[Tooltip](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/tooltip) component does not work with disabled [Button](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/button) component:

**Steps to reproduce:**

```jsx
/**
 * External dependencies
 */
import { Tooltip, Button } from '@wordpress/components';

const MyComponent = () => {
	return (
		<Tooltip text="Show helpful text here">
			<Button disabled>Click me</Button>
		</Tooltip>
	);
};

export default MyComponent;
```

1. Create a React component like the above MyComponent.
1. Use the component in your React page. 
1. Open up a browser, navigate to the React page, hover your mouse over the disabled "Click me" button. **Expect the tooltip to show up, but it does not.**
1. Remove the `disabled` prop, and reload the page. The tooltip will show up as expected.

**Expected behavior**

Tooltip should appear when I hover over the disabled button.

### PR DESCRIPTION
**Pull request by grzim**
## Description

Closes https://github.com/WordPress/gutenberg/issues/27451
According to the issue https://github.com/WordPress/gutenberg/issues/27451 - there was no mechanism for disabled elements to catch events and to display a tooltip. 
I have added a layer placed above disabled elements that work as an event catcher and therefore make a tooltip work.

I wouldn't say it is a bug fix, rather a change in current behavior. In my opinion, this change would benefit UX.

## How has this been tested?

Unit test added to cover this scenario. 
Manual test.

## Screenshots 
alt text

## Types of changes
Current behaviour altered

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/tooltip/index.js [hunk 1]
```diff
@@ -24,7 +24,65 @@ import Shortcut from '../shortcut';
...
  * @type {number}
  */
-const TOOLTIP_DELAY = 700;
+export const TOOLTIP_DELAY = 700;
+
+const eventCather = (
+	<div
+		className="event-catcher"
+		style={ {
+			zIndex: 9999,
```
Comment (draganescu) [inline packages/components/src/tooltip/index.js:33]: why is this inline styling better than using a class name and normal CSS?

Comment (draganescu) [inline packages/components/src/tooltip/index.js:33]: We have afaik a z-index management SCSS file with variables instead if hardcoded values.

File: packages/components/src/tooltip/test/index.js [hunk 2]
```diff
@@ -4,7 +4,10 @@
 import { shallow, mount } from 'enzyme';
 import TestUtils from 'react-dom/test-utils';
 import ReactDOM from 'react-dom';
-
+/**
+ * Internal dependencies
+ */
+import { TOOLTIP_DELAY } from '../index.js';
```
Comment (draganescu) [inline packages/components/src/tooltip/test/index.js:10]: Do we need to export and import this? can't we define one for the test alone?

File: packages/components/src/tooltip/index.js [hunk 3]
```diff
@@ -24,7 +24,65 @@ import Shortcut from '../shortcut';
...
+			width: '100%',
+			height: '100%',
+			top: 0,
+		} }
+	/>
+);
+
+const getDisabledElement = ( { eventHandlers, child, childrenWithPopover } ) =>
+	cloneElement(
+		<span style={ { position: 'relative' } }>
```
Comment (draganescu) [inline packages/components/src/tooltip/index.js:44]: Same comment with inline style as above. I may have missed this kind of styling, is it also used in other components?

**Review by draganescu — COMMENTED** [review]
Due to this being a change in interaction at a component level I'll also add two apropiate labels.

File: packages/components/src/tooltip/test/index.js (hunk 2 above)
Comment (grzim) [reply packages/components/src/tooltip/test/index.js:10]: I think it is better to use the same constant in both tests and the code itself as it is is a constant

Comment (draganescu) [reply packages/components/src/tooltip/test/index.js:10]: does this by being exported become a part of the public API?

Comment (grzim) [reply packages/components/src/tooltip/test/index.js:10]: it is not exported in components/src/index.js so I believe it is not public.

**Comment by jameskoster** [comment]
I'm not sure if this is caused by the overlay, or an issue with the buttons inherent disabled state (if so feel free to ignore), but I don't think the appearance should change on hover if the button is disabled.

**Comment by grzim** [comment]
Nice catch! I have checked it and this behaviour is not connected to an overlay itself.

**Comment by jameskoster** [comment]
In that case I don't think this really needs a design review since nothing has changed visually.

**Comment by shaunandrews** [comment]
> In my opinion, this change would benefit UX.

Allowing disabled buttons to display a tooltip makes a lot of sense to me.

File: packages/components/src/tooltip/index.js [hunk 4]
```diff
@@ -104,7 +151,7 @@ function Tooltip( { children, position, text, shortcut } ) {
 			}
 
 			// Needed in case unsetting is over while delayed set pending, i.e.
-			// quickly blur/mouseleave before delayedSetIsOver is called
+			// quickly blur/mouseleave before delayedSetIsOver is cgualled
```
Comment (draganescu) [inline packages/components/src/tooltip/index.js:154]: Small typo 👍

**Review by draganescu — CHANGES_REQUESTED** [review]

**Review by draganescu — APPROVED** [review]
Tested this and looks great. Code lgtm 🥇

**Pull request merged**